test/executor.test.js

```diff
@@ -352,41 +386,51 @@ describe('executor', function() {
       delete process.env.npm_package_config_port;
     });
 
-    function bootWithDefaults() {
+    function bootWithDefaults(done) {
       app = loopback();
       boot.execute(app, someInstructions({
-        config: {
+        application: {
           port: undefined,
           host: undefined,
         },
-      }));
+      }), done);
     }
 
-    it('should apply env passed in option object', function() {
-      boot.execute(app, someInstructions({ env: 'custom_env' }));
-      expect(app.get('env')).to.equal('custom_env');
+    it('should apply env passed in option object', function(done) {
+      boot.execute(app, someInstructions({ env: 'custom_env' }), function(err) {
+        if (err) return done(err);
+        expect(app.get('env')).to.equal('custom_env');
+        done();
+      });
     });
 
-    it('should honor host and port', function() {
-      function assertHonored(portKey, hostKey) {
+    it('should honor host and port', function(done) {
+      function assertHonored(portKey, hostKey, done) {
```

Comment:
Similarly here, it may be confused there are two `done` callback. I am proposing to use `cb` or `callback` as the argument name in `assertHonored`.

Actually, even better solution is to rework this unit test into a parameterized test case.

```js
[
  { port: 'OPENSHIFT_SLS_PORT', host: 'OPENSHIFT_NODEJS_IP' },
  // ..
].forEach(function(tc) {
  it('should honor ' + tc.host + ' and ' + tc.port, function(done) {
    var hostKey = tc.host;
    var portKey = tc.key;
    // copy the code from assertHonored here
  });
});
```

That way we continue running tests for other ENV variables even after one of them was detected as not working. The output of `mocha -R spec` will also nicely describe what different ENV variables are supported.